packages/glib/build.sh

```diff
@@ -67,11 +66,62 @@ termux_step_host_build() {
 termux_step_pre_configure() {
 	# glib checks for __BIONIC__ instead of __ANDROID__:
 	CFLAGS+=" -D__BIONIC__=1"
+	_PREFIX=$TERMUX_PKG_TMPDIR/prefix
+	local _WRAPPER_BIN="${TERMUX_PKG_BUILDDIR}/_wrapper/bin"
+	rm -rf $_PREFIX $_WRAPPER_BIN
+	mkdir -p $_PREFIX $_WRAPPER_BIN
 
-	TERMUX_PKG_VERSION=. termux_setup_gir
+	sed '/^export PKG_CONFIG_LIBDIR=/s|$|:'${_PREFIX}'/lib/pkgconfig|' \
+		"${TERMUX_STANDALONE_TOOLCHAIN}/bin/pkg-config" \
+		> "${_WRAPPER_BIN}/pkg-config"
+	chmod +x "${_WRAPPER_BIN}/pkg-config"
+	export PKG_CONFIG="${_WRAPPER_BIN}/pkg-config"
+	export PATH="${_WRAPPER_BIN}:${PATH}"
+
+	# Magic happens here.
+	# I borrowed nested building method from https://github.com/termux/termux-packages/blob/1244c75380beefc7f7da9744d55aa88df1640acb/x11-packages/qbittorrent/build.sh#L21-L28
+	# and modified termux_step_configure_meson in runtime to make it use another prefix
+	# Also I used advice from here https://github.com/termux/termux-packages/issues/20447#issuecomment-2156066062
+
+	(
+		eval `declare -f termux_step_configure_meson | sed 's|\$TERMUX_PREFIX|'$_PREFIX'|g'`
+		TERMUX_PKG_BUILDDIR=$TERMUX_PKG_TMPDIR/glib-build
+		mkdir -p $TERMUX_PKG_BUILDDIR
+		TERMUX_PKG_EXTRA_CONFIGURE_ARGS="${TERMUX_PKG_EXTRA_CONFIGURE_ARGS/"-Dintrospection=enabled"/"-Dintrospection=disabled"}"
+		TERMUX_PKG_VERSION=. termux_setup_gir
+		
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_configure
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_make
+		cd "$TERMUX_PKG_BUILDDIR"
+		termux_step_make_install
+	)
+
+	(
+		eval `declare -f termux_step_configure_meson | sed 's|\$TERMUX_PREFIX|'$_PREFIX'|g'`
+		TERMUX_PKG_BUILDER_DIR=$TERMUX_SCRIPTDIR/packages/gobject-introspection
+		TERMUX_PKG_BUILDDIR=$TERMUX_PKG_TMPDIR/gobject-introspection-build
+		TERMUX_PKG_SRCDIR=$TERMUX_PKG_TMPDIR/gobject-introspection-src
+		mkdir -p $TERMUX_PKG_BUILDDIR $TERMUX_PKG_SRCDIR
+		. $TERMUX_PKG_BUILDER_DIR/build.sh
```

Comment:
So, we're re-sourcing the build script recursively here?
That definitely does the trick.

I think using the `source` keyword would make that clearer though.
Also if you could write a quick comment to explain that, that would be great.
```suggestion
		# Blood sacrifices or something probably
		source $TERMUX_PKG_BUILDER_DIR/build.sh
```

Reply:
`source` keyword is used only in `ncdu2` and `luvit` packages, in other places (`tsmuxergui`, `tigervnc`, `mpv`, `mpv-x` and tens of others) `.` is used for sourcing buildscripts. And they are not commented.

Reply:
`.` for sourcing is a personal pet peeve of mine.
I shouldn't bring it up as a criticism in reviews, sorry.

Please do still add a comment about what the purpose of recursively sourcing the build script is.

Reply:
I am not sure how to comment this line. `# Shave Yaks` is the best explanation I can imagine.

Reply:
And again, it does not have comment in other places.

Reply:
> And again, it does not have comment in other places.

Are we re-sourcing the same build script anywhere else inside itself?

> I am not sure how to comment this line. `# Shave Yaks` is the best explanation I can imagine.

Sorry, yak shaving is an idiom for a task that doesn't seem to have any immediate relation with the stated goal.
https://en.wiktionary.org/wiki/yak_shaving

If you'd prefer a visual example, see this skit.
https://youtu.be/_UZFI-8D5uA

Reply:
> Are we re-sourcing the same build script anywhere else inside itself?

No. sourcing is done in a subshell (see parentheses) to not interfere with the rest of the script.

> If you'd prefer a visual example, see this skit.

But I still do not know what to write in comment :( .

Reply:
> No. sourcing is done in a subshell (see parentheses) to not interfere with the rest of the script.

That might be good to point out because that is a quirk of Bash that is not readily apparent.